### DISCUSSION TITLE
Excessive upgrade when only 1 upgrade was requested?

### DISCUSSION THREAD

**Discussion by ghoffart** [Homebrew/discussions#327] [Everyday usage]
Hi,

I’m doing a `brew install sysbench`, and minutes later I wonder why it still isn’t finished: it’s because homebrew is upgrading a **ton** of other packages. First I thought these are dependencies, but this is not the case: it’s just outdated packages of my system - which I didn’t want to upgrade for a reason.

How can I avoid having unrequested upgrades?

Homebrew 2.6.2-134-g24046c3, but happens since a few months. Environment: macOS Catalina 10.15.7 on Intel (x64_64 version), and Big Sur 11.1 on M1 (using the x86_64 version, still)

**Comment by jonchang** [comment]
See: https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-also-upgrade-a-bunch-of-other-stuff

**Reply by ghoffart** [reply]
Thanks, but the upgrades I see are _IMHO_ unrelated to the required action. I installed 'sysbench' and got 'tn5250', 'poppler' and 'postgresql' upgraded. I don’t think these depend on each other.

**Reply by iMichka** [reply]
> brew deps sysbench
mysql-client
openssl@1.1

> brew uses --recursive openssl@1.1
postgresql, poppler

So they are related.

For tn5250 I do not know, we do not provide this formula.

**Reply by gromgit** [reply]
Odds are good that wherever @ghoffart got the `tn5250` formula, it also depends on `openssl@1.1`.

**Reply by carlocab** [reply]
Homebrew used to. It's even got an `:el_capitan` bottle. 👀

https://bintray.com/homebrew/bottles/tn5250#

**Comment by tlk** [comment]
> How can I avoid having unrequested upgrades?

I noticed in another discussion that it is possible to disable homebrew auto updates:
```bash
echo "export HOMEBREW_NO_AUTO_UPDATE=1" >> ~/.zshrc
```

https://github.com/Homebrew/brew/blob/603a0599cb3d4ba4a42dcc145770b187bad91212/Library/Homebrew/env_config.rb#L201-L203

**Reply by carlocab** [reply]
Setting that makes it so that `brew update` is not run before `brew install` or `brew upgrade`. It will not stop `brew upgrade foo` from also upgrading, say, `bar`, if `bar` also needed upgrading. So this isn't quite what the OP is looking for.
